docs/_spec/05-classes-and-objects.md

```diff
@@ -989,106 +984,119 @@ transparent trait Enum extends Any, Product, Serializable:
 ```
 There are nine desugaring rules.
 Rule (1) desugars enum definitions.
-Rules (2) and (3) desugar simple cases.
-Rules (4) to (6) define `extends` clauses for cases that are missing them.
-Rules (7) to (9) define how such cases with `extends` clauses map into `case class`es or `val`s.
+Rule (2) desugars cases of comma-separated names to simple enum cases.
+Rules (3) to (7) desugar inferrable details of enum cases.
+Rules (8) and (9) define how fully-desugared enum cases map into `case class`es or `val`s.
+Explicit `extends` clauses must be provided in the following cases, where rules (2) to (6) do not apply:
+- any enum case of a parameterized enum,
+- any singleton enum case of an unparameterized enum with non-variant type parameters,
+- any class enum case of an enum with type parameters, where the case also has type parameters.
 
 1.  An `enum` definition
     ```scala
-    enum ´E´ ... { <defs> <cases> }
+    enum ´E´ <type-params> <value-params> { <defs> <cases> }
     ```
     expands to a `sealed abstract` class that extends the `scala.reflect.Enum` trait and an associated companion object that contains the defined cases, expanded according to rules (2 - 8).
     The enum class starts with a compiler-generated import that imports the names `<caseIds>` of all cases so that they can be used without prefix in the class.
     ```scala
-    sealed abstract class ´E´ ... extends <parents> with scala.reflect.Enum {
-        import ´E´.{ <caseIds> }
-        <defs>
+    sealed abstract class ´E´ <type-params> <value-params>
+        extends <parents> with scala.reflect.Enum {
+      import ´E´.{ <caseIds> }
+      <defs>
     }
     object ´E´ { <cases> }
     ```
 
-2. A singleton case consisting of a comma-separated list of enum names
+2. A simple enum case consisting of a comma-separated list of names
    ```scala
    case ´C_1´, ..., ´C_n´
    ```
-   expands to
+   expands to the following simple enum cases
    ```scala
    case ´C_1´; ...; case ´C_n´
    ```
    Any modifiers or annotations on the original case extend to all expanded cases.
-   This result is then further rewritten by either (3 or 4).
+   <p>This result is then further rewritten by either (3 or 4).</p>
```

Comment:
Does that actually render in a meaningful way?

Reply:
I thought it looks nicer with the clean break, there's no way to do that in pure markdown without breaking the numbered list